wp-parsely.php

```diff
@@ -721,7 +721,29 @@ public function insert_parsely_page() {
 
 		$current_url = $this->get_current_url();
 
-		if ( in_array( get_post_type(), $parsely_options['track_post_types'], true ) && 'publish' === $post->post_status ) {
+		if ( is_home() ) {
+			$parsely_page['@type'] = 'WebPage';
+			// If front page is set to display a static page, get the URL of the posts page.
```

Comment:
I'm having a hard time visualizing these two situations and the urls they will return. I'm sure I'm missing something obvious about why we can't just use `get_home_url` for both. Docstring examples would be nice, but test cases are probably preferable.

Reply:
I'm not sure how to explain the situations, but what I can say is this:
for a page where it's using home.php as a posts page (in the Readings settings), and with us having code of:
```
// If front page is set to display a static page, get the URL of the posts page.
            if ( 'page' === get_option( 'show_on_front' ) ) {
                $parsely_page['url'] = get_permalink( get_option( 'page_for_posts' ) );
            } else {
                $parsely_page['url'] = get_home_url();
            }
```
in place, we get:
```
<script type="application/ld+json">
    {"@context":"http:\/\/schema.org","@type":"WebPage","url":"http:\/\/vccw.test\/front-page","headline":"Welcome to the VCCW"}
</script>
```
(note the url, which in this case is the *correct* url for the page we're on)

but if our code is using `get_home_url()` like:
```
$parsely_page['@type'] = 'WebPage';
            $parsely_page['url'] = get_home_url();
```
then the metadata we get is:
```
<script type="application/ld+json">
    {"@context":"http:\/\/schema.org","@type":"WebPage","url":"http:\/\/vccw.test","headline":"Welcome to the VCCW"}
</script>
```
(note the url, which in this case is *not* the correct url - it's just the base url of the home page, but not the page we're on that has all the posts).

I _think_ this has to do with using home.php as a template for a page that is listing posts, as opposed to operating as a static page, but basically it's something I googled because I wasn't getting the correct url, and this was the solution, and it seems to work?

Obviously this is exactly the kind of thing we need to put unit tests in place for, but this is an example anyway of why we can't just use `get_home_url` for both situations.

Does this make sense given your WP experience? I'm kind of fumbling in the dark, so if there's a better approach, I am totally open to it!

Reply:
I _think_ I understand now. Tell me if this is sounds right: in the `is_home()` check, wordpress isn't really checking if you're on the homepage (website root); it's checking whether you're on a page generated by the `home.php` template. In contrast, `get_home_url()` always just returns the root url of the site.

Assuming that's right, I understand why ``get_home_url()` won't work. But now that makes me think functions like `get_home_url()` and `get_permalink( get_option( 'page_for_posts' ) )` are both equally dangerous, in that they rest on assumptions of relationships that may not be true (like that the homepage template will always _be_ the homepage). I think what we want is a more generic function that simply gets the url we're currently on.

A little bit of research led to `home_url( $wp->request )`, which, ugh, seems like a horrible hack.

Reply:
@joshuarrrr I think you've got it right (and put it into words way better than I could, so thank you!). I agree, too, that those function calls seem suspect due to the assumptions they're making. Because indeed, in the case of millionmilesecrets.com, they're using home.php but not for the actual homepage, so a more generic function would be great... even if it's a horrible hack. I'd rather have a hack that works than something more "elegant" that doesn't!

Reply:
yeah, mostly, I'm just annoyed that there seems to be no standard function that just gets the current link - like, c'mon wordpress!

Reply:
If we wanted to be ultra thorough about it, I guess we could look at the source code for the official AMP plugin, which also needs to generate ld+json.